src/digitalmarketplace/components/list-input/template.njk

```diff
@@ -0,0 +1,110 @@
+{%- from "../../../govuk-frontend/components/error-message/macro.njk" import govukErrorMessage -%}
+{%- from "../../../govuk-frontend/components/fieldset/macro.njk" import govukFieldset -%}
+{%- from "../../../govuk-frontend/components/hint/macro.njk" import govukHint -%}
+{%- from "../../../govuk-frontend/components/label/macro.njk" import govukLabel -%}
+{%- from "../../../govuk-frontend/components/input/macro.njk" import govukInput -%}
+
+{#-  We need this for error messages and hints as well -#}
+{%- set id = params.id -%}
+
+{#- A record of elements we want associated with the fieldset's aria-describedby -#}
+{%- set describedBy = params.fieldset.describedBy -%}
+
+{%- if params.hint %}
+  {%- set hintId = id + '-hint' -%}
+  {%- set describedBy = describedBy + ' ' + hintId if describedBy and hintId not in describedBy else hintId -%}
+{% endif -%}
+
+{%- if params.errorMessage %}
+  {%- set listErrorId = id + '-error' -%}
+  {%- set describedBy = describedBy + ' ' + listErrorId if describedBy and listErrorId not in describedBy else listErrorId -%}
+{% endif -%}
+
+<div class="dm-list-input{% if params.classes %} params.classes {% endif %}{% if params.errorMessage %} govuk-form-group--error{% endif %}">
+  {% call govukFieldset({
+    describedBy: params.fieldset.describedBy if params.fieldset.describedBy else describedBy,
+    classes: params.fieldset.classes,
+    legend: {
+      html: params.fieldset.legend.text + ('<span class="dm-list-input__optional__optional"> (Optional)</span>' if params.optional) ,
+      classes: 'dm-list-input__legend ' + ( params.fieldset.legend.classes if params.fieldset.legend.classes)
+    },
+    attributes: {
+      'data-module': 'dm-list-input'
+    }
+  }) %}
+
+  {%- if params.hint %}
+    {{ govukHint({
+        id: hintId,
+        classes: 'dm-list-input__hint ' + ( params.hint.classes if params.hint.classes),
+        attributes: params.hint.attributes,
+        html: params.hint.html,
+        text: params.hint.text
+      }) | indent(2) | trim() }}
+  {% endif -%}  
+  
+  {% if params.errorMessage %}
+    {{ govukErrorMessage({
+      id: listErrorId,
+      classes: 'dm-list-input-error-message' + params.errorMessage.classes if params.errorMessage.classes,
+      attributes: params.errorMessage.attributes,
+      html: params.errorMessage.html,
+      text: params.errorMessage.text,
+      visuallyHiddenText: params.errorMessage.visuallyHiddenText
+    }) | indent(2) | trim }}
+  {% endif %}
+
+  <div class="dm-list-input__item-container" role="region" aria-live="assertive">
+    {% for i in range(0, params.maxItems) -%}
+      {%- set item = params.items[i] if params.items[i] else {} -%}
+      {%- set entryItemId = params.id + '-' + i -%}
```

Comment:
We should use `loop.index` here the same way that govuk-frontend does so that the id numbers are 1 indexed as other components (for an example see [checkboxes component](https://github.com/alphagov/govuk-frontend/blob/master/src/govuk/components/checkboxes/template.njk#L61-L66))